packages/core/contracts/financial-templates/perpetual-multiparty/PerpetualLiquidatable.sol

```diff
@@ -225,8 +227,8 @@ contract PerpetualLiquidatable is PerpetualPositionManager {
         external
         fees()
         notEmergencyShutdown()
-        updateFundingRate()
         nonReentrant()
+        updateFundingRate()
```

Comment:
Why do this? This might actually throw if its after `nonReentrant`

Reply:
Yeah, this is probably not the right way to fix this, but the updateFundingRate is _not_ currently protected by a reentrancy guard because it isn't a public method. So the quick fix was to put it inside the re-entrancy guard modifier whenever used. But this means we have an inconsistent setup where some modifiers are protected on their own and some need to be placed after the re-entrancy guard.

Reply:
What do you think?

Reply:
I think its fine as long as you leave a comment because I could see this easily confusing us in the future and leading to strange test breakage

Reply:
i.e. "placing this modifier after the reentrancy-guard modifier protects its internal methods against reentrancy"

Reply:
Okay, will do. I'll consider fixing this in a more elegant way in a follow-up, but for now, I'll add a comment since I agree that the subtlety is confusing.